core/sds-aspect-model-document-generators/src/main/java/io/openmanufacturing/sds/aspectmodel/generator/jsonschema/AspectModelJsonSchemaVisitor.java

```diff
@@ -239,18 +242,38 @@ public JsonNode visitHasProperties( final HasProperties element, final ObjectNod
       return context;
    }
 
+   private Characteristic determineCharacteristic( final Property property ) {
+      if ( property.getCharacteristic() != null ) {
```

Comment:
In which scenario is the Characteristic `null`?

Reply:
The Characteristic can be `null` if the Property is abstract.

Reply:
Would it not make sense to define the Characteristic as optional in the `Property` class?

Reply:
You're correct. I have now updated `getCharacteristic()` in Property to return `Optional<Characteristic>`. This communicates in a clearer way that there are cases in which no Characteristic is present.